### PR TITLE
Ensure trim() is always used on string value and not on array type

### PR DESCRIPTION
**Pull request by mariadanieldeepak**
Fix https://github.com/sudar/email-log/issues/171

### CODE REVIEW

File: include/Core/DB/TableManager.php [hunk 1]
```diff
@@ -315,9 +315,9 @@ public function fetch_log_item_by_item_data( $data ) {
 		// Execute the following `if` conditions only when $data is array.
 		if ( array_key_exists( 'to', $data ) ) {
 			// Since the value is stored as CSV in DB, convert the values from error data to CSV to compare.
-			$data['to'] = Util\join_array_elements_with_delimiter( $data['to'] );
+			$to_email = Util\join_array_elements_with_delimiter( $data['to'] );
 
-			$to_email = trim( esc_sql( $data['to'] ) );
+			$to_email = trim( esc_sql( $to_email ) );
```
Comment (sudar) [inline include/Core/DB/TableManager.php]: Note from Scrutinizer

`It seems like esc_sql($to_email) can also be of type array; however, parameter $str of trim() does only seem to accept string, maybe add an additional type check?`

File: include/Util/helper.php [hunk 2]
```diff
@@ -129,7 +129,7 @@ function join_array_elements_with_delimiter( $value, $delimiter = ',' ) {
 		return implode( $delimiter, $value );
 	}
 
-	return $value;
+	return is_string( $value ) ? $value : '';
```
Comment (sudar) [inline include/Util/helper.php]: Note from Scrutinizer

`The condition is_string($value) is always true.`

File: include/Core/DB/TableManager.php (hunk 1 above)
Comment (mariadanieldeepak) [reply include/Core/DB/TableManager.php]: @sudar Though the note has already been handled, yet we need to figure out a way to suppress  the message.

This line https://github.com/sudar/email-log/blob/171-fix/scrutinizer-issue/include/Core/DB/TableManager.php#L318 will always return a string and not an array.

File: include/Util/helper.php (hunk 2 above)
Comment (mariadanieldeepak) [reply include/Util/helper.php]: The same note applies here too.

**Pull request merged**